### PR TITLE
Enhanced search: enforce query criteria via `require_all_on` argument

### PR DESCRIPTION
**Pull request by andersy005**
Towards #202 

Cc @jbusecke, @matt-long

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -255,3 +255,9 @@ def test_get_subset():
 
     assert set(x.keys()) == set(['CESM', 'FOO'])
     assert list(x.values()) == [2, 2]
+
+    x = _get_subset(df, C=['control', 'hist'], D=['NO2'], force_all_on='B')
```
Comment (matt-long) [inline tests/test_core.py]: Would `require_all_on` be a better name than `force_all_on`? I think it would.

Comment (andersy005) [reply tests/test_core.py]: Done!

**Comment by andersy005** [comment]
I am planning on merging this later this afternoon unless there is an objection and/or additional feedback.

**Comment by jbusecke** [comment]
Testing right now. Will report back in a bit.

**Pull request merged**